tests/transformers/fnet/test_modeling.py

```diff
@@ -0,0 +1,291 @@
+        result = model(input_ids, return_dict=True)
+        self.parent.assertEqual(
+            result["last_hidden_state"].shape, [self.batch_size, self.seq_length, self.hidden_size]
+        )
+
+    def create_and_check_for_pretraining(
+        self, config, input_ids, token_type_ids, sequence_labels, token_labels, choice_labels
+    ):
+        model = FNetForPreTraining(config=config)
+        model.eval()
+        result = model(
+            input_ids,
+            token_type_ids=token_type_ids,
+            labels=token_labels,
+            next_sentence_label=sequence_labels,
+            return_dict=True,
+        )
+        self.parent.assertEqual(result["prediction_logits"].shape, [self.batch_size, self.seq_length, self.vocab_size])
+        self.parent.assertEqual(result["seq_relationship_logits"].shape, [self.batch_size, 2])
+
+    def create_and_check_for_masked_lm(
+        self, config, input_ids, token_type_ids, sequence_labels, token_labels, choice_labels
+    ):
+        model = FNetForMaskedLM(config=config)
+        model.eval()
+        result = model(input_ids, token_type_ids=token_type_ids, labels=token_labels, return_dict=True)
+        self.parent.assertEqual(result["prediction_logits"].shape, [self.batch_size, self.seq_length, self.vocab_size])
+
+    def create_and_check_for_next_sentence_prediction(
+        self, config, input_ids, token_type_ids, sequence_labels, token_labels, choice_labels
+    ):
+        model = FNetForNextSentencePrediction(config=config)
+        model.eval()
+        result = model(
+            input_ids,
+            token_type_ids=token_type_ids,
+            next_sentence_label=sequence_labels,
+            return_dict=True,
+        )
+        self.parent.assertEqual(result["logits"].shape, [self.batch_size, 2])
+
+    def create_and_check_for_question_answering(
+        self, config, input_ids, token_type_ids, sequence_labels, token_labels, choice_labels
+    ):
+        model = FNetForQuestionAnswering(config=config)
+        model.eval()
+        result = model(
+            input_ids,
+            token_type_ids=token_type_ids,
+            start_positions=sequence_labels,
+            end_positions=sequence_labels,
+            return_dict=True,
+        )
+        self.parent.assertEqual(result["start_logits"].shape, [self.batch_size, self.seq_length])
+        self.parent.assertEqual(result["end_logits"].shape, [self.batch_size, self.seq_length])
+
+    def create_and_check_for_sequence_classification(
+        self, config, input_ids, token_type_ids, sequence_labels, token_labels, choice_labels
+    ):
+        config.num_labels = self.num_labels
+        model = FNetForSequenceClassification(config)
+        model.eval()
+        result = model(input_ids, token_type_ids=token_type_ids, labels=sequence_labels, return_dict=True)
+        self.parent.assertEqual(result["logits"].shape, [self.batch_size, self.num_labels])
```

Comment:
不建议使用键值对的方式获取模型的输出（虽然可行）
```suggestion
        self.parent.assertEqual(result.logits.shape, [self.batch_size, self.num_labels])
```